### PR TITLE
add search autocomplete

### LINKED ISSUE #1098: Update search to use OSM data
# Problem
Currently search data is not using osm data

# Solution
1. Create API to API V2 to be able to search
2. Search in here are : Healthsites suggestion & Place suggestion. (When clicked into 1 of healthsite suggestion, ignore this issue because i'm working on it)
3. update static to use require and backbone

child of https://github.com/healthsites/healthsites/issues/1116

### PR DESCRIPTION
**Pull request by ann26**
fix #1098 

Peek 2019-04-16 11-57

### CODE REVIEW

File: django_project/api/api_views/v2/locality_osm_autocomplete.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# coding=utf-8
+__author__ = 'Anita Hapsari <anita@kartoza.com>'
+__date__ = '15/04/19'
+
+import json
+from django.http import HttpResponse
+from localities_osm.models.locality import LocalityOSMView
+
+
+def locality_autocomplete(request):
```
Comment (meomancer) [inline django_project/api/api_views/v2/locality_osm_autocomplete.py]: hi @ann26 
please put this into v2/localities
and rename this file into search.py

also please rename function into search_by_name and use APIView

File: django_project/api/api_views/v2/locality_osm_autocomplete.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    q = request.GET.get('q', '').capitalize()
+    if len(q) > 2:
```
Comment (meomancer) [inline django_project/api/api_views/v2/locality_osm_autocomplete.py]: please control when len less than 2 (maybe give error 400)

File: django_project/api/api_views/v2/locality_osm_autocomplete.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        search_qs = LocalityOSMView.objects.filter(
+            name__icontains=q)
+        results = []
+        for r in search_qs:
```
Comment (meomancer) [inline django_project/api/api_views/v2/locality_osm_autocomplete.py]: please use serializer, maybe "LocalityOSMAutocomplete" class

File: django_project/api/api_views/v2/locality_osm_autocomplete.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+import json
+from django.http import HttpResponse
+from localities_osm.models.locality import LocalityOSMView
+
+
+def locality_autocomplete(request):
+    q = request.GET.get('q', '').capitalize()
+    if len(q) > 2:
+        search_qs = LocalityOSMView.objects.filter(
```
Comment (meomancer) [inline django_project/api/api_views/v2/locality_osm_autocomplete.py]: please use get_all_osm_query

File: django_project/api/api_views/v2/locality_osm_autocomplete.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            name__icontains=q)
+        results = []
+        for r in search_qs:
+            results.append({
+                'pk': r.pk,
+                'text': r.name,
+            })
+        data = json.dumps(results)
+        mime_type = 'application/json'
+        return HttpResponse(data, mime_type)
```
Comment (meomancer) [inline django_project/api/api_views/v2/locality_osm_autocomplete.py]: please use Response of rest framework

File: django_project/api/urls.py [hunk 6]
```diff
@@ -30,6 +31,8 @@
         GetDetailFacility.as_view(), name='api_v2_facility_detail'),
     url(r'^facilities',
         GetFacilities.as_view(), name='api_v2_facility_list'),
+    url(r'^search-name/',
```
Comment (meomancer) [inline django_project/api/urls.py]: change into "facilities/autocomplete"

**Comment by meomancer** [comment]
Also please change css of autocomplete, using black color with bold for "searched words"

**Comment by ann26** [comment]
done @meomancer 
Peek 2019-04-16 12-28

File: django_project/api/api_views/v2/localities/search.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    def get(self, request):
+        q = request.GET.get('q', '').capitalize()
+        if len(q) > 2:
+            query = get_all_osm_query()
+            search_qs = query.filter(name__icontains=q)
+            serializer = \
+                LocalityOSMAutoCompleteSerializer(search_qs, many=True)
+            return Response(serializer.data)
+        else:
+            return Response(
```
Comment (meomancer) [inline django_project/api/api_views/v2/localities/search.py]: please use HttpBadRequest

**Comment by meomancer** [comment]
great, thank you @ann26 
i will continue to update js using require

**Comment by meomancer** [comment]
thank you @ann26 
i'm sorry i force push your this PR :(

**Pull request merged**